src/millie.nim

```diff
@@ -10,46 +10,41 @@ const units = @[
   "P", # quadrillion
 ]
 
-## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
+
 proc millify*(
     value: int64,
     precision: int = 1,
     lowercase: bool = false,
     space: bool = false,
   ): string =
+  ## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
   
-  # validate if `vaue` can be parsed or not
-  var val: SomeNumber = parseNum(value)
+  # preallocate the resulting string
+  result = newStringOfCap(10)
 
-  # `prefix` will be used later when assembling
-  let prefix: string = if val < 0: "-" else: ""
+  var val = float(value)
 
+  # add a negative prefix if needed
+  if val < 0: result.add '-'
+  
   # only work with positive numbers for easiness
   val = val.abs()
 
   # keep dividing the input value by the digit grouping base (1000)
   # until the number and the unit index are found.
-  var unitIndex: int = 0
-  proc disposableFunc(res: float) =
+  var unitIndex = 0
+  divideTillEnd(val):
```

Comment:
I've changed `divideTillEnd` to be a template instead of a procedure - this way we can pass the code we need to execute inside it, and there's no function call overhead at all.